tests/kani/UnsizedCoercion/custom_outer_coercion.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: --mir-linker --enable-unstable
+//! Check the basic coercion when using a custom CoerceUnsized implementation.
+//! Tests are broken down into different crates to ensure that the reachability works for each case.
+#![feature(coerce_unsized)]
+#![feature(unsize)]
+use std::marker::Unsize;
+use std::ops::{CoerceUnsized, Deref};
+
+mod defs;
+use defs::*;
+
+/// Dummy reference wrapper that allow unsized coercion.
+pub struct MyPtr<'a, T: ?Sized> {
+    ptr: &'a T,
+}
+
+impl<'a, T: ?Sized + Unsize<U>, U: ?Sized> CoerceUnsized<MyPtr<'a, U>> for MyPtr<'a, T> {}
```

Comment:
I kept misunderstanding was was meant by `CoerceUnsized`. Can we add a link to https://doc.rust-lang.org/std/ops/trait.CoerceUnsized.html here :)

Reply:
Sure